modules/programs/gradle.nix

```
@@ -0,0 +1,89 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.gradle;
+  settingsFormat = pkgs.formats.javaProperties { };
+in {
+  meta.maintainers = [ maintainers.britter ];
+
+  options.programs.gradle = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+    };
+    home = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = ''
+        The Gradle home directory.
```

Since this is managed by `home.file`, it’ll have to be a relative path:

```suggestion
        The Gradle home directory, relative to `home.homeDirectory`.
```